### PR TITLE
8270517: Add Zero support for LoongArch

### PR DESCRIPTION
**Pull request by xiangzhai**
Hi,

I'd like to backport this patch to jdk8u. `common/autoconf/build-aux/config.guess` and `hotspot/src/os/linux/vm/os_linux.cpp` do not apply cleanly due to context difference, but it is easy to resolve them manually. `common/autoconf/platform.m4` just changed file path.

`common/autoconf/generated-configure.sh` regenerated by `bash common/autoconf/autogen.sh`.

A native build on LoongArch hardware is tested.

Debian:  https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1060236

```
$ ./build/images/j2sdk-image/bin/java -version
openjdk version "1.8.0_392"
OpenJDK Runtime Environment (build 1.8.0_392-8u392-ga-1-b08)
OpenJDK 64-Bit Zero VM (build 25.392-b08, interpreted mode)
```

Loongnix Desktop:

```
$ ./build/linux-loongarch64-normal-zero-release/images/j2sdk-image/bin/java -version
openjdk version "1.8.0_412-internal"
OpenJDK Runtime Environment (build 1.8.0_412-internal-zhaixiang_2024_01_06_10_53-b00)
OpenJDK 64-Bit Zero VM (build 25.412-b00, interpreted mode)
```

The risk of the downport is low.

Thanks,
Leslie Zhai

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8270517](https://bugs.openjdk.org/browse/JDK-8270517) needs maintainer approval

### Issue
 * [JDK-8270517](https://bugs.openjdk.org/browse/JDK-8270517): Add Zero support for LoongArch (**Enhancement** - P4 - Approved)

### Reviewers
 * [Ao Qi](https://openjdk.org/census#aoqi) (@theaoqi - no project role)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/413/head:pull/413` \
`$ git checkout pull/413`

Update a local copy of the PR: \
`$ git checkout pull/413` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/413/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 413`

View PR using the GUI difftool: \
`$ git pr show -t 413`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/413.diff">https://git.openjdk.org/jdk8u-dev/pull/413.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/413#issuecomment-1878412061)

### CODE REVIEW

**Comment by theaoqi** [comment]
As the author of [8270517](https://bugs.openjdk.org/browse/JDK-8270517), the backport looks good to me. The releaes and fastdebug native builds are passed on my loongarch64 platform. The GHA test failure seems unrelated, see https://github.com/openjdk/jdk8u-dev/pull/169#issuecomment-1353434077 and https://github.com/openjdk/jdk8u-dev/pull/228#issuecomment-1380658336.

**Review by theaoqi — APPROVED** [review]

**Comment by xiangzhai** [comment]
Hi @theRealAph 

Could you sponsor it?

Thanks,
Leslie Zhai

**Comment by xiangzhai** [comment]
Hi @jerboaa

Could you review and sponsor it please?

Thanks,
Leslie Zhai

**Comment by theaoqi** [comment]
> Hi @jerboaa
> 
> Could you review and sponsor it please?
> 
> Thanks, Leslie Zhai

I think you should get a review (by a Reviewer of jdk8u) and the maintainer's approval first.

**Review by gnu-andrew — APPROVED** [review]
This looks right to me. Does this allow Zero build on LoongArch?

**Comment by gnu-andrew** [comment]
To move this forward, you need to formally request maintainer approval.  See https://wiki.openjdk.org/display/SKARA/Pull+Request+Commands#PullRequestCommands-/approval

I'm assuming from the thumbs up that it does allow Zero builds on LoongArch :)

**Comment by xiangzhai** [comment]
/approval request allow Zero build on LoongArch

**Comment by mrserb** [comment]
It is strange but after integration of this patch, execution of "autogen.sh" using autoconf 2.69 will cut part of the change(tested on macOS+brew).

[autoconf (GNU Autoconf) 2.69]
Generating generated-configure.sh
```diff
diff --git a/common/autoconf/generated-configure.sh b/common/autoconf/generated-configure.sh
index 85eb8a16a21..0803a8782af 100644
--- a/common/autoconf/generated-configure.sh
+++ b/common/autoconf/generated-configure.sh
@@ -1017,7 +1017,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -1263,7 +1262,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
@@ -1516,15 +1514,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1662,7 +1651,7 @@ fi
 for ac_var in  exec_prefix prefix bindir sbindir libexecdir datarootdir \
                datadir sysconfdir sharedstatedir localstatedir includedir \
                oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-               libdir localedir mandir runstatedir
+               libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1815,7 +1804,6 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -4440,7 +4428,7 @@ VS_TOOLSET_SUPPORTED_2022=true
```

**Comment by mrserb** [comment]
the same result is on Linux, the next run of "autogen.sh" will delete part of this change.
@xiangzhai please take a look.

**Comment by xiangzhai** [comment]
Hi @mrserb 

Due to [AC_INIT: add --runstatedir option to configure](https://git.savannah.gnu.org/gitweb/?p=autoconf.git;a=commit;h=a197431414088a417b407b9b20583b2e8f7363bd).  And I also tested with autoconf 2.69 and **2.72**, there is `runstatedir` generated.

Thanks,
Leslie Zhai

**Comment by zifeihan** [comment]
> > Hi @mrserb
> > Due to [AC_INIT: add --runstatedir option to configure](https://git.savannah.gnu.org/gitweb/?p=autoconf.git;a=commit;h=a197431414088a417b407b9b20583b2e8f7363bd). And I also tested with autoconf 2.69 and **2.72**, there is `runstatedir` generated.
> > Thanks, Leslie Zhai
> 
> Yes, we've gone back and forth on this before. Some distros have this locally patched in, others don't, and it really depends where `configure` is generated. I think I may have even generated both forms myself on different systems.
> 
> I'm thinking of just backporting [JDK-8195689](https://bugs.openjdk.org/browse/JDK-8195689) to remove the generated `configure` at the start of the next release cycle. The argument for not having `autoconf` installed was weak to begin with and is now limited to those who don't have to work on any later JDKs, which all have this change. I think it causes more confusion than it helps at this point and there is always the risk of someone forgetting to regenerate it.

@gnu-andrew : Hi, May I ask if there is a plan to backport [JDK-8195689](https://bugs.openjdk.org/browse/JDK-8195689) ? Thanks.

**Pull request closed without merging**
